### ISSUE TITLE
Report field-Thread for report field system message is grayed out when expense is not submitted

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#52242]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.59-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**Email or phone of affected tester (no customers):** applausetester+kh2710000@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
Precondition:
- Workspace has some report fields.

1. Go to staging.new.expensify.com
2. Go to workspace chat > + > Submit expense.
3. Submit the manual expense (do not click Submit button in the expense preview).
4. Go to expense report.
5. Click on the report field.
6. Edit the report field and save it.
7. Right click on the report field system message > Reply in thread (if the message does not appear, refresh the page).
8. Note that the thread is grayed out and composer is hidden.
9. Clear cache and restart the app, or log out and log in
10. Go back to expense report.
11. Now the report field system message disappears.

## Expected Result:
In Step 8. the report field system message thread should not be grayed out and composer will not be hidden.

## Actual Result:
In Step 8. the report field system message thread is grayed out and composer is hidden.
In Step 11, the system message disappears after clear cache and restart the app, or relogin.

## Workaround:
Unknown

## Platforms:

- [x] Android: Standalone
- [x] Android: HybridApp
- [x] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/5a06471c-c85f-4375-b46f-1fb29c00cfab

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021858401001231358475</li>
        <li>Upwork Job ID: 1858401001231358475</li>
        <li>Last Price Increase: 2025-02-17</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @robertjchen</details>

**Comment by truph01** [comment]
It is BE bug. There is no ```CHANGEFIELD``` action returned by BE.

**Comment by VictoriaExpensify** [comment]
I haven't had a chance to look at this but will come back to it tomorrow

**Comment by VictoriaExpensify** [comment]
I agree this is an issue, especially with composer being hidden, we should fix this.

**Comment by mkzie2** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
In Step 8. the report field system message thread is grayed out and composer is hidden.
In Step 11, the system message disappears after clear cache and restart the app, or relogin.

### What is the root cause of that problem?
The backend will not return any report action if we edit the report field if the report has not been submitted. However, we always create the optimistic change field action without checking whether the expense report has been submitted.

So the thread is invalid and the action disappears after we reset cache and restart

https://github.com/Expensify/App/blob/d5e7a5abde0dfc52312a15feba776f3ef2f0be49/src/libs/actions/Report.ts#L2020-L2025
### What changes do you think we should make in order to solve the problem?
We should only add an optimistic change field action if the expense report is submitted

```
    if (!ReportUtils.isOpenExpenseReport(expenseReport)) {
        const optimisticChangeFieldAction = ReportUtils.buildOptimisticChangeFieldAction(reportField, previousReportField);
        optimisticData.push(
            {
                onyxMethod: Onyx.METHOD.MERGE,
                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
                value: {
                    [optimisticChangeFieldAction.reportActionID]: optimisticChangeFieldAction,
                },
            },
        )

        failureData.push(
            {
                onyxMethod: Onyx.METHOD.MERGE,
                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
                value: {
                    [optimisticChangeFieldAction.reportActionID]: {
                        errors: ErrorUtils.getMicroSecondOnyxErrorWithTranslationKey('report.genericUpdateReportFieldFailureMessage'),
                    },
                },
            },
        )
    }
```

https://github.com/Expensify/App/blob/d5e7a5abde0dfc52312a15feba776f3ef2f0be49/src/libs/actions/Report.ts#L2020-L2025

### What alternative solutions did you explore? (Optional)

**Comment by mkzie2** [comment]
@robertjchen To clarify, the report field is still updated but the backend doesn't return the change field action if the report isn't submitted. Can you please check if it's expected from the backend or not? 

- If not, this is an internal issue and we need to fix it from the backend side.

- If the backend is correct, we should fix it as a front-end bug.

**Comment by mkzie2** [comment]
@robertjchen Can you help to check [this](https://github.com/Expensify/App/issues/52242#issuecomment-2500007827)?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by robertjchen** [comment]
Will need to look in the backend, but it seems like it's no longer reproducible?

**Comment by VictoriaExpensify** [comment]
Correct @robertjchen. Let me know if you see anything in the backend, otherwise I think we can close this out

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by VictoriaExpensify** [comment]
Hey @robertjchen - it seems @mkzie2 can still reproduce this. Have you had a chance to take a look at the backend?

**Comment by VictoriaExpensify** [comment]
@parasharrajat engineers self-assign to the issues so I can't just reassign to someone else. I'll reach out to @robertjchen directly

**Comment by VictoriaExpensify** [comment]
Just realised that Robert is probably missing the pings because the issue wasn't starred to him. I've done this now so it should appear in his queue

**Comment by robertjchen** [comment]
👀

**Comment by robertjchen** [comment]
to be handled internally

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week)

**Comment by robertjchen** [comment]
Thanks, closing for now due to non-reproducibility during KI tests
